tests/testthat/test-class-Megastudy.R

```diff
@@ -544,7 +599,15 @@ test_that("imputeZeroes method is sane", {
       variableSpec = new("VariableSpec", variableId = 'attractant', entityId = 'collection'),
       plotReference = new("PlotReference", value = 'facet1'),
       dataType = new("DataType", value = 'STRING'),
-      dataShape = new("DataShape", value = 'CATEGORICAL'))
+      dataShape = new("DataShape", value = 'CATEGORICAL')),
+    new("VariableMetadata",
+      variableClass = new("VariableClass", value = 'native'),
+      variableSpec = new("VariableSpec", variableId = 'sex', entityId = 'sample'),
+      # empty plotReference means that it is not plotted
+      dataType = new("DataType", value = 'STRING'),
+      dataShape = new("DataShape", value = 'CATEGORICAL'),
+      weightingVariableSpec = VariableSpec(variableId='specimen_count',entityId='sample'),
+      hasStudyDependentVocabulary = TRUE)
```

Comment:
Is the error expected because `sex` is repeated or because we asked for a mix of sample variables with and without `hasStudyDependentVocabulary`?

Presumably would we have used `sample.eyecolor` here if we had it.

Reply:
not sure tbh. i think maybe i had a stroke while writing this one lol

Reply:
ok. i removed the duplicate sample.sex reference in the VariableMetadataList. I now expect this to fail bc smaple.sex is present, and claims to have a study specific vocab, but no vocab was provided for it in the megastudy obj